next/13/replace-next-head-v2/test.ts

```diff
@@ -117,18 +105,106 @@ describe('next 13 replace-next-head-v2', function () {
 			data:
 				'import { Metadata } from "next";\n' +
 				"import Meta from '../../components/a.tsx';\n" +
-				"import { e1, e } from 'module1';\n" +
-				'const c = () => { };\n' +
-				'const { b } = { b: "content" };\n' +
-				'const a = "content";\n' +
 				'export const metadata: Metadata = {\n' +
-				'    title: `${title}`,\n' +
-				'    applicationName: a + b + c() + d() + e,\n' +
-				'    description: description,\n' +
+				'    title: `title`,\n' +
+				'    description: "description",\n' +
+				'};\n' +
+				'export default function Page() {\n' +
+				'    return <Meta />;\n' +
+				'}\n',
+		});
+	});
+
+	it('should move definitions of identifiers used in meta tag expr to the Page file', async function (this: Context) {
+		const A_CONTENT = `
+		import Meta from '../../components/a.tsx';
+		export default function Page() {
+			return <Meta />;
+		}
+`;
+
+		const A_COMPONENT_CONTENT = `
+		import Head from 'next/head';
+		
+		const a = "a";
+		const b = () => "b";
+		function c() { return "c" };
+		const { obj: { d }} = { obj: { d: "d"} };
+		const env = process.env.APP_NAME;
+		
+		export default function Meta() {
+			return (<>
+			<Head>
+				<title>{a + b() + c() + d + e + env}</title>
+			</Head>
+			</>)
+		}
+`;
+
+		const externalFileCommands = await transform({
+			'/opt/project/pages/a/index.tsx': A_CONTENT,
+			'/opt/project/components/a.tsx': A_COMPONENT_CONTENT,
+			'/opt/project/utils/index.ts': '',
+		});
+
+		deepStrictEqual(externalFileCommands[0], {
+			kind: 'upsertFile',
+			path: '/opt/project/pages/a/index.tsx',
+			data:
```

Comment:
I was thinking we could use backticks and remove the whitespaces with a regex, similar to other tests, e.g.:
```
const data = `
			'use client';
			import ErrorPage from 'next/error';
			
			const NotFound = () => <ErrorPage statusCode={404} />;
			
			export default NotFound;
		`;

		deepStrictEqual(
			upsertNotFoundFileCommand?.data.replace(/\W/gm, ''),
			data.replace(/\W/gm, ''),
		);
```

Reply:
sure